src/cpp/src/lm_encoding.cpp

```diff
@@ -175,6 +169,13 @@ ov::genai::utils::GenerationFinishInfo get_lm_encoded_results(
     SamplerOutput sampler_output = sampler.sample(sequence_groups, logits);
     free_non_running_requests(); // handle sampler output
 
+    const auto infer_end = std::chrono::steady_clock::now();
+    const auto infer_ms = PerfMetrics::get_microsec(infer_end - infer_start);
+    raw_perf_counters.m_inference_durations[0] += MicroSeconds(infer_ms);
+    raw_perf_counters.m_token_infer_durations.emplace_back(infer_ms);
+    raw_perf_counters.m_new_token_times.emplace_back(infer_end);
+    raw_perf_counters.m_batch_sizes.emplace_back(sampler_output.num_generated_tokens);
```

Comment:
That and below code block movement increases measured inference duration. I think you should call `now()` twice: after infer and here `m_new_token_times`. Why would otherwise we have two entries in `RawPerfMetrics`